http/codegen/service_data.go

```diff
@@ -1444,7 +1495,7 @@ func buildBodyType(svc *service.Data, s *httpdesign.ServiceExpr, e *httpdesign.E
 		// Otherwise when generating types that are used to unmarshal
 		// use pointer fields regardless of whether the attribute is
 		// required or has a default value.
-		marshaled = !svr && req || svr && !req
+		marshaled = !svr && req
```

Comment:
Is `marshaled` still useful like this? would it be better to leave it (would be used on line 1538) and then add a new `needValidation` that would be used on line 1521?